symless/synergy-core
# Simplify compile options and use comprehensive edition logic

https://symless.atlassian.net/browse/S1-1698

To test:
1. Configure without extra args
2. Expect: Title says 'Community Edition'
1. Configure with `-DENABLE_LICENSING=ON`
3. Test a serial key
4. Expect: Serial key is accepted
